qiskit/providers/aer/noise/errors/quantum_error.py

```diff
@@ -292,367 +400,68 @@ def to_dict(self):
-            raise NoiseError(
-                "QuantumErrors are not defined on same number of qubits.")
-
-        combined_noise_circuits = []
-        combined_noise_probabilities = []
-
-        # Combine subcircuits and probabilities
-        if left_multiply:
-            noise_ops0 = list(
-                zip(self._noise_circuits, self._noise_probabilities))
-            noise_ops1 = list(
-                zip(other._noise_circuits, other._noise_probabilities))
-        else:
-            noise_ops0 = list(
-                zip(other._noise_circuits, other._noise_probabilities))
-            noise_ops1 = list(
-                zip(self._noise_circuits, self._noise_probabilities))
-        # Combine subcircuits and probabilities
-        for circuit0, prob0 in noise_ops0:
-            for circuit1, prob1 in noise_ops1:
-                combined_noise_probabilities.append(prob0 * prob1)
-                tmp_combined = circuit0 + circuit1
-
-                # Fuse compatible ops to reduce noise operations:
-                combined_circuit = [tmp_combined[0]]
-                for instr in tmp_combined[1:]:
-                    last_instr = combined_circuit[-1]
-                    last_name = last_instr['name']
-                    name = instr['name']
-                    can_combine = (last_name in ['id', 'kraus', 'unitary'] or
-                                   name in ['id', 'kraus', 'unitary'])
-                    if (can_combine and self._check_instr(last_name) and
-                            self._check_instr(name)):
-                        combined_circuit[-1] = self._compose_instr(
-                            last_instr, instr, self.number_of_qubits)
-                    else:
-                        # If they cannot be combined append the operation
-                        combined_circuit.append(instr)
-                # Check if circuit is empty and add identity
-                if not combined_circuit:
-                    combined_circuit.append({'name': 'id', 'qubits': [0]})
-                # Add circuit
-                combined_noise_circuits.append(combined_circuit)
-        noise_ops = self._combine_kraus(
-            zip(combined_noise_circuits, combined_noise_probabilities),
-            self.number_of_qubits)
-        return QuantumError(noise_ops)
-
-    def _tensor_product(self, other, reverse=False):
-        """Return the tensor product error channel.
 
-        Args:
-            other (QuantumError): a quantum channel subclass
-            reverse (bool): If False return self ⊗ other, if True return
-                            if True return (other ⊗ self) [Default: False
-        Returns:
-            QuantumError: the tensor product error channel.
+        if self.num_qubits != other.num_qubits:
+            raise NoiseError("Number of qubis of other ({}) must be the same as self ({})".format(
+                other.num_qubits, self.num_qubits))
```

Comment:
Although I kept this check (respecting backward compat), it might be not required any more (we could enlarge circuits automatically). What do you think? Should we leave it as is for now?

Reply:
The new `compose()` implementation can handle composition of errors with different number of qubits, so I decided to remove this error check at 5d85966.